alternator/streams.cc

```diff
@@ -690,7 +685,7 @@ future<executor::request_return_type> executor::get_shard_iterator(client_state&
             inclusive_of_threshold = false;
             break;
         case shard_iterator_type::TRIM_HORIZON:
-            threshold = utils::UUID{};
+            threshold = utils::UUID_gen::min_time_UUID(sid->time.time_since_epoch().count());
```

Comment:
Why is this change needed? In case by freak chance we reuse the same stream token for a new generation as was already used in an older generation?

Reply:
Actually, on second thought, this should probably go. I put it there to handle potentially changing to range-based-shards eventually, but it is probably better to just retain the zero-uuid threshold so we don't miss any data in the time of generation overlap.